addons/hr/models/res_users.py

```diff
@@ -238,34 +285,27 @@ def write(self, vals):
             for field_name, field in self._fields.items()
             if field.related_field and field.related_field.model_name == 'hr.employee' and field_name in vals
         }
+        user_initial_values = defaultdict(dict)
+
         can_edit_self = self.env['ir.config_parameter'].sudo().get_param('hr.hr_employee_self_edit') or self.env.user.has_group('hr.group_hr_user')
         if hr_fields and not can_edit_self:
             # Raise meaningful error message
             raise AccessError(_("You are only allowed to update your preferences. Please contact a HR officer to update other information."))
+        elif hr_fields and can_edit_self:
+            for user in self:
+                for field in hr_fields.keys():
+                    user_initial_values[user][field] = user.sudo()[field]
 
         employee_domain = [
             *self.env['hr.employee']._check_company_domain(self.env.company),
             ('user_id', 'in', self.ids),
         ]
-        if hr_fields:
-            employees = self.env['hr.employee'].sudo().search(employee_domain)
-            get_field = self.env['ir.model.fields']._get
-            field_names = Markup().join([
-                 Markup("<li>%s</li>") % get_field("res.users", fname).field_description for fname in hr_fields
-            ])
-            for employee in employees:
-                reason_message, partner_ids = self._get_personal_info_partner_ids_to_notify(employee)
-                if partner_ids:
-                    employee.message_notify(
-                        body=Markup("<p>%s</p><p>%s</p><ul>%s</ul><p><em>%s</em></p>") % (
-                            _('Personal information update.'),
-                            _("The following fields were modified by %s", employee.name),
-                            field_names,
-                            reason_message,
-                        ),
-                        partner_ids=partner_ids,
-                    )
         result = super(User, self).write(vals)
+        if bool(user_initial_values):
+            user_ids = self._get_users_to_notify()
+            if can_edit_self and user_ids:
```

Comment:
Hello @robcoekaerts 
we want to send notifications only when self-edit is enabled from the employee settings
so I've modified the if condition
Thank you for looking into it :innocent:

Reply:
Hi, my interpretation is that if someone edits their own data, the notification should (potentially) be sent, even if the reason the person can edit their own data is that they are an hr user. So I don't think the 'employee_edit_self' field is needed. 

Instead, to prevent notifications from being sent when the self-edit option is disabled (and payroll is not installed) I'd add an if check to `def _get_users_to_notify(self):` where you check whether edit self is enabled or not, to prevent sending the notification to `hr_notify_user_ids`

Reply:
Hello @robcoekaerts 
I've relocated the if condition :innocent: